src/main/java/org/generationcp/middleware/dao/ProjectDAO.java

```diff
@@ -1,180 +1,256 @@
+				query.addEntity("w", Project.class);
+				query.setParameter("userId", userId);
+
+				@SuppressWarnings("unchecked")
+				final List<Project> projectList = query.list();
+
+				return !projectList.isEmpty() ? projectList.get(0) : null;
+			}
+		} catch (final HibernateException e) {
+			throw new MiddlewareQueryException("Error with getLastOpenedProject(userId=" + userId + ") query from Project " + e.getMessage(), e);
+		}
+		return null;
+	}
+
+	public Project getLastOpenedProjectAnyUser() throws MiddlewareQueryException {
+		try {
+
+			final StringBuilder sb = new StringBuilder();
+			sb.append("SELECT {w.*} FROM workbench_project w ")
+					.append("INNER JOIN workbench_project_user_info r ON w.project_id = r.project_id ")
+					.append("WHERE r.last_open_date IS NOT NULL ORDER BY r.last_open_date DESC LIMIT 1 ;");
+
+			final SQLQuery query = this.getSession().createSQLQuery(sb.toString());
+			query.addEntity("w", Project.class);
+
+			@SuppressWarnings("unchecked")
+			final List<Project> projectList = query.list();
+
+			return !projectList.isEmpty() ? projectList.get(0) : null;
+
+		} catch (final HibernateException e) {
+			throw new MiddlewareQueryException("Error with getLastOpenedProjectAnyUser(" + ") query from Project " + e.getMessage(), e);
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	public List<Project> getProjectsByCrop(final CropType cropType) throws MiddlewareQueryException {
+		final Criteria criteria = this.getSession().createCriteria(Project.class).add(Restrictions.eq("cropType", cropType));
+		return criteria.list();
+	}
+
+	public List<Project> getProjectsByFilter(final int pageNumber,final int pageSize, final Map<ProgramFilters, Object> filters)
+		throws MiddlewareException {
+		try {
+			final Criteria criteria = this.getSession().createCriteria(Project.class);
+			for (Map.Entry<ProgramFilters, Object> entry : filters.entrySet()) {
+				final ProgramFilters filter = entry.getKey();
+				final Object value = entry.getValue();
+				criteria.add(Restrictions.eq(filter.getStatement(), value));
+			}
+
+			final int start = pageSize * (pageNumber - 1);
+			final int numOfRows = pageSize;
+
+			criteria.setFirstResult(start);
+			criteria.setMaxResults(numOfRows);
+			return criteria.list();
+		} catch (HibernateException e) {
+			throw new MiddlewareException(
+				"Error in getProjectsByFilter(start=" + pageNumber + ", numOfRows=" + pageSize + "): " + e.getMessage(), e);
+		}
+	}
+
+	public long countProjectsByFilter(final Map<ProgramFilters, Object> filters) throws MiddlewareException {
+		try {
+			final Criteria criteria = this.getSession().createCriteria(Project.class);
+			for (Map.Entry<ProgramFilters, Object> entry : filters.entrySet()) {
+				final ProgramFilters filter = entry.getKey();
+				final Object value = entry.getValue();
+				criteria.add(Restrictions.eq(filter.getStatement(), value));
+			}
+
+			return criteria.list().size();
+		} catch (HibernateException e) {
```

Comment:
Apply code cleanup (eg. append final to applicable variables, parameters) to ProjectDao.java